cli/command/cli.go

```diff
@@ -32,27 +32,42 @@ type Streams interface {
 // DockerCli represents the docker command line client.
 // Instances of the client can be returned from NewDockerCli.
 type DockerCli struct {
-	configFile      *configfile.ConfigFile
-	in              *InStream
-	out             *OutStream
-	err             io.Writer
-	keyFile         string
-	client          client.APIClient
-	hasExperimental *bool
+	configFile            *configfile.ConfigFile
+	in                    *InStream
+	out                   *OutStream
+	err                   io.Writer
+	keyFile               string
+	client                client.APIClient
+	hasExperimental       bool
+	originalClientVersion string
+}
+
+// Ping pings the server and retrieve the version to use and if experimental is enabled.
+func (cli *DockerCli) Ping() (err error) {
```

Comment:
`Ping()` is maybe not the best name for this, how about making this an unexpected func that is called from `Initialize()` ?  `initializeVersion()` ?